app/dals/post_dal.py

```diff
@@ -1,69 +1,98 @@
-from app.exceptions import UserNotFoundException
+from app.exceptions import OperationError, PostNotFoundException
 from app.DB.db_operations import DatabaseOperations
-from app.exceptions import PostNotFoundException
 from ..DB.models import Post
+from .queries_statement.query_params import posts_statement_by_name
+from ..pydantic_models.post_models.post_request_model import *
+from ..pydantic_models.post_models.post_response_model import *
+from uuid import UUID
 
-async def get_all_posts() -> list:
+
+async def get_all_posts(feedReqs: PostFeedRequestModel) -> list[PostGetResponseModel]:
     print("Getting all posts...")
+    posts_response_list = []
     try:
         db_operations = DatabaseOperations()
         with db_operations.get_session() as session:
             print("Getting all posts...")
-            posts = session.query(Post).all()
+            query = session.query(Post).filter(Post.isActive == True)
+
+            for filter_name, value in feedReqs.__dict__.items():
+                print("Filter name in join: ", filter_name)                
+                query = posts_statement_by_name[filter_name].append_join(query)
+            
+            for filter_name, condition in feedReqs.__dict__.items():
```

Comment:
For sure it has a better way to iterate the key, values of a dict instead of `feedReqs.__dict__`